### ISSUE TITLE
No shadow for framework version of AsyncTaskLoader

### ISSUE THREAD

**Issue by rumpeltux** [robolectric/robolectric#1582]
When trying to test an Activity/Fragment that uses an AsyncTaskLoader, the loader is run and completes successfully, but the `onLoadFinished` method is never called.
(FTR: I’m not using the support library but the framework implementation of AsyncTaskLoader)

Is that a bug in Robolectric or an issue with test setup?
See also http://stackoverflow.com/questions/18280752/testing-asynctaskloaders-with-robolectric

**Comment by rumpeltux** [comment]
I think there’s two parts to it. I didn’t call .start() on the activityController.

The other is that there’s no non-support-library shadow implementation of AsyncTaskLoader so that the AsyncTask will always be run on a real worker thread.

**Comment by jrodbx** [comment]
Check Robolectric.getUiThreadScheduler() and see if your task is enqueued.  If so, you can call runOneTask(), etc. which should indirectly invoke onLoadFinished().

**Comment by rumpeltux** [comment]
Yes, though that only works after adding a shadow implementation of android.content.AsyncTaskLoader (robolectric only includes one for android.support.v4.content.AsyncTaskLoader).

**Comment by dingzhihu** [comment]
i added `org.robolectric.shadows.ShadowAsyncTaskLoader` when testing my `Activity`, but `android.support.v4.content.AsyncTaskLoader's` `onLoadFinished` still not called.

**Comment by jaredsburrows** [comment]
@rumpeltux  What is the status of this? Did you get it to work?

**Comment by rumpeltux** [comment]
Nope, I just gave up ;)

**Comment by jongerrish** [comment]
Another scheduling quirk: @kriegfrj @erd 

This breaks if you call LoaderManager.destroyLoader() from 

LoaderManager.LoaderCallbacks.onLoadFinished(Loader<Cursor> loader, Cursor cursor)

because the LoaderManager is still in the "creating" stage because the callbacks are executed eagerly by the unpaused looper inside this block, rather than outside the block as they would be on real android.

https://github.com/android/platform_frameworks_base/blob/kitkat-mr2-release/core/java/android/app/LoaderManager.java#L556

java.lang.RuntimeException: java.lang.IllegalStateException: Called while creating a loader
	at org.robolectric.util.SimpleFuture.run(SimpleFuture.java:61)
	at org.robolectric.shadows.ShadowAsyncTaskLoader$2.run(ShadowAsyncTaskLoader.java:46)
	at org.robolectric.util.Scheduler.runOrQueueRunnable(Scheduler.java:230)
	at org.robolectric.util.Scheduler.postDelayed(Scheduler.java:85)
	at org.robolectric.util.Scheduler.post(Scheduler.java:72)
	at org.robolectric.shadows.ShadowAsyncTaskLoader.onForceLoad(ShadowAsyncTaskLoader.java:43)
	at android.content.AsyncTaskLoader.onForceLoad(AsyncTaskLoader.java)
	at android.content.Loader.forceLoad(Loader.java:347)
	at android.content.CursorLoader.onStartLoading(CursorLoader.java:158)
	at android.content.Loader.startLoading(Loader.java:290)
	at android.app.LoaderManagerImpl$LoaderInfo.start(LoaderManager.java:280)
	at android.app.LoaderManagerImpl.installLoader(LoaderManager.java:569)
	at android.app.LoaderManagerImpl.createAndInstallLoader(LoaderManager.java:556)
	at android.app.LoaderManagerImpl.initLoader(LoaderManager.java:609)

**Comment by kriegfrj** [comment]
My proposed fix for #1306 should take care of this but it is dependent on some of the work in #1879.

**Comment by koral--** [comment]
Maybe it should be separate issue but I'm first writing here.
I'm using robolectric 3.0 with the following config:
```
@RunWith(RobolectricGradleTestRunner.class)
@Config(constants = BuildConfig.class, application = TestApp.class, sdk = Build.VERSION_CODES.LOLLIPOP, packageName = "com.myapp")
```
Activity is obtained this way: `Robolectric.buildActivity(MyActivity.class).setup().get()`.

I'm using `android.content.AsyncTaskLoader` and `onLoadFinished()` is still not called. Loading is successful and I see that `Loader.deliverResult()` is called on real object but `mListener` is `null` so nothing happens.

Is it bug in robolectric or something more in test code is required for `AsyncTaskLoader` to fully work?
